ga4gh/datamodel/__init__.py

```diff
@@ -229,6 +256,8 @@ class VariantSetCompoundId(DatasetCompoundId):
     """
     fields = DatasetCompoundId.fields + ['variantSet']
     containerIds = DatasetCompoundId.containerIds + [('variantSetId', 1)]
+    obfuscators = [DatasetCompoundId, None]
```

Comment:
Each type has its own obfuscator.  Obfuscating the same string with two different obfuscators will return two different strings by very virtue of there needing to be globally unique IDs.  CompoundIds store IDs of parent objects as attributes on themselves.  In order to do this, they need access to the correct type obfuscators.